presto-docs/src/main/sphinx/develop/release-process.rst

```diff
@@ -0,0 +1,179 @@
+
+**Automated Testing**
+   * Unit tests on all commits and RCs
+   * Product tests with full cluster deployment
+   * Shared connector tests via `AbstractTestQueries <https://github.com/prestodb/presto/blob/master/presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java>`_
+   * CI for basic stability checks
+
+**Testing Contributions Needed**
+   * Additional `product test scenarios <https://github.com/prestodb/presto/tree/master/presto-product-tests>`_
+   * Performance regression testing (shadow traffic)
+   * Test coverage for new features
+
+Version Numbering
+-----------------
+
+* Format: ``0.XXX`` (e.g., 0.293, 0.294)
+* Major version fixed at 0
+* Minor version increments each release
+* Patch releases (e.g., 0.293.1) for critical fixes only
+
+Not semantic versioning.
+
+Release Types
+-------------
+
+**Regular Releases**
+   * Every ~2 months
+   * New features, improvements, bug fixes
+   * Extended RC testing period
+
+**Patch Releases**
+   * Critical issues only:
+     
+     - Security vulnerabilities (upon request - default: upgrade)
+     - Data correctness
+     - Performance regressions
+     - Stability issues
+   
+   * Based on previous stable release
+   * Minimal changes
+
+**Release Candidates**
+   * 2-4 week testing period
+   * One RC per release
+   * Fixes verified in existing RC
+
+**Edge Releases**
+   * Weekly from master
+   * Early access to features
+   * Testing only - not for production
+
+Release Shepherd Responsibilities
+----------------------------------
+
+Requirements:
+
+* Must be a committer
+* Must understand codebase for judgment calls and rewrites
+
+Responsibilities:
+
+* Complete release process
+* Ensure release notes follow guidelines
+* Fix release note issues
+* Cut and deploy release
+* Coordinate with community
+* Make go/no-go decisions
+
+See `Release Shepherding wiki <https://github.com/prestodb/presto/wiki/Release-Shepherding>`_ for schedule and details.
+
+Contributing to Release Quality
+-------------------------------
+
+**Testing**
+   * Run RCs in test environments
+   * Report issues with reproduction steps
+   * Verify cherry-picked fixes
+
+**Test Development**
+   * Add product tests for uncovered scenarios
+   * Unit tests for bug fixes
+   * Performance benchmarks (`pbench <https://github.com/prestodb/pbench>`_)
+
+**Documentation**
+   * Document behavior changes
+   * Write clear release notes
+   * Note compatibility issues
+
+**Code Review**
+   * Review PRs for correctness
+   * Identify compatibility issues
+   * Suggest test coverage
+
+Backward Compatibility Guidelines
+----------------------------------
+
+**Must Maintain Compatibility:**
+
+* **Client Libraries**: Evolve slowly. New server features must work with older clients.
+* **SQL Syntax**: Keep stable. Deprecate with warnings before removal.
+* **SPI**: Stable for connectors/plugins. Use ``@Deprecated`` for several releases before removal.
```

Comment:
We haven't been very good about keeping the SPI stable, so maybe should spell out more explicit guidance.  For example, when adding new SPI methods prefer having reasonable defaults, so most connectors don't need to be updated unless they want to support the new feature.